extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Specialized;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Newtonsoft.Json;
+using System.Linq;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    /// <summary>
+    /// Converter to bind Blob Storage type parameters.
+    /// </summary>
+    internal class BlobStorageConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (context.Source is not IModelBindingData bindingData)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            if (bindingData.Source is not Constants.BlobExtensionName)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var content = GetBindingDataContent(bindingData);
+            var connectionName = content.FirstOrDefault(x => String.Equals(x.Key, Constants.Connection, StringComparison.OrdinalIgnoreCase)).Value;
+            var containerName = content.FirstOrDefault(x => String.Equals(x.Key, Constants.ContainerName, StringComparison.OrdinalIgnoreCase)).Value;
+            var blobName = content.FirstOrDefault(x => String.Equals(x.Key, Constants.BlobName, StringComparison.OrdinalIgnoreCase)).Value;
+
+            if (string.IsNullOrEmpty(connectionName) || string.IsNullOrEmpty(containerName))
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            var connectionString = connectionName is null ? null : Environment.GetEnvironmentVariable(connectionName);
```

Comment:
I don't thing we should be defaulting the connection here. There's logic in the blob extension itself that is creating the ParameterBindingData, and we should centralize all the connection name resolution/defaulting there I think.